feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go

```diff
@@ -537,18 +580,281 @@ func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good, bad, lb []
 		}
 	}
 	for _, flow := range bad {
-		if got := gnmi.Get(t, ate, gnmi.OC().Flow(flow.Name()).LossPct().State()); got < 100 {
-			t.Fatalf("LossPct for flow %s: got %g, want 100", flow.Name(), got)
+		if !change {
+			if got := gnmi.Get(t, ate, gnmi.OC().Flow(flow.Name()).LossPct().State()); got < 100 {
+				t.Fatalf("LossPct for flow %s: got %g, want 100", flow.Name(), got)
+			}
+		}
+	}
+
+	if change {
+		var receivedPkts uint64
+
+		incoming_traffic_counters := gnmi.OC().Interface(ate.Port(t, "port1").Name()).Counters()
+		sentPkts := gnmi.Get(t, ate, incoming_traffic_counters.OutPkts().State())
+
+		// Get traffic received on primary outgoing interface before modifying NHG
+		for _, port := range nonrx_ports {
+			outgoing_traffic_counters := gnmi.OC().Interface(ate.Port(t, port).Name()).Counters()
+			outPkts := gnmi.Get(t, ate, outgoing_traffic_counters.InPkts().State())
+			receivedPkts = receivedPkts + outPkts
+		}
+
+		// Get traffic received on expected port after modifying NHG
+		for _, outPort := range expected_outgoing_port {
+			outgoing_traffic_counters := gnmi.OC().Interface(outPort.Name()).Counters()
+			outPkts := gnmi.Get(t, ate, outgoing_traffic_counters.InPkts().State())
+			receivedPkts = receivedPkts + outPkts
+		}
+
+		// Check if traffic restores with in expected time in milliseconds during modify NHG
+		if len(nonrx_ports) > 0 {
+			// Time took for traffic to restore in milliseconds after trigger
+			fpm := ((sentPkts - receivedPkts) / (fps / 1000))
```

Comment:
i dont see fps it set while creatingFlow() so how can we make sure ATE is sending only with fps rate that is used here.

Reply:
Have added frame size and fps to the flows.